### PR TITLE
Coupon modal

### PR DESCRIPTION
**Pull request by Developers-codz**
## files to check

`src/App.js`
`src/component/modal/Modal.jsx`
`src/context/cart-context.js`
`src/pages/cart/Cart.jsx`

Also added proper navigation after login
Thank you for reviewing the PR :)

### CODE REVIEW

File: src/pages/cart/Cart.jsx [hunk 1]
```diff
@@ -80,6 +82,14 @@ const Cart = () => {
       </div>
       <div className="cart-right-pane">
         <div className="bill-card">
+          <h4 class="mb-lg text-vibrant">COUPONS</h4>
```
Comment (hsnice16) [inline src/pages/cart/Cart.jsx:85]: instead of `class`, use `className`.

File: src/context/cart-context.js [hunk 2]
```diff
@@ -119,6 +130,31 @@ const CartProvider = ({ children }) => {
       console.log(err);
     }
   };
+  const total = cartState.cartData.reduce(
+    (acc, curr) => acc + curr.price * curr.qty,
+    0
+  );
+  console.log(total);
```
Comment (hsnice16) [inline src/context/cart-context.js]: if this `console` statement is not required, you can remove this.

File: src/component/modal/Modal.jsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    setCouponDiscount,
+  } = useCart();
+  const total = cartData.reduce((acc, curr) => acc + curr.price * curr.qty, 0);
+  return (
+    <div
+      class="coupon-modal"
+      id="coupon-popup"
+      style={isModalOpen ? { display: "block" } : { display: "none" }}
+    >
+      <div class="modal-close-btn" onClick={() => setModalOpen(false)}>
```
Comment (hsnice16) [inline src/component/modal/Modal.jsx]: use `className`.

File: src/component/ProtectedRoute.jsx [hunk 4]
```diff
@@ -1,12 +1,15 @@
...
 const ProtectedRoute = () => {
-
   const {
     authState: { isAuthTokenPresent },
   } = useAuth();
-  // console.log(isAuthTokenPresent);
-  return isAuthTokenPresent ? <Outlet /> : <Navigate to="/login" />;
+  const location = useLocation();
+  console.log(location);
+  return isAuthTokenPresent ? (
```
Comment (vishal804) [inline src/component/ProtectedRoute.jsx:9]: I guess this Console.log is for testing Purpose It can be removed if not needed

**Review by vishal804 — APPROVED** [review]

**Comment by vishal804** [comment]
Code Looks clean and readable

**Review by jalajcodes — APPROVED** [review]

File: src/component/NonAuthenticatedRoute.jsx [hunk 5]
```diff
@@ -1,11 +1,16 @@
...
+import { Navigate, Outlet, useLocation } from "react-router-dom";
 import { useAuth } from "context/auth-context";
-import { Profile } from "pages";
 
 export const NonAuthenticatedRoute = () => {
   const {
     authState: { isAuthTokenPresent },
   } = useAuth();
+  const location = useLocation();
+  console.log(location);
```
Comment (hsnice16) [inline src/component/NonAuthenticatedRoute.jsx]: remove this `console`, if it is not required.

**Review by hsnice16 — APPROVED** [review]
Nice work. Have added a few suggestions.

File: src/pages/cart/Cart.jsx (hunk 1 above)
Comment (Developers-codz) [reply src/pages/cart/Cart.jsx:85]: Thats by mistake, Thanks for pointing that out himanshu

File: src/context/cart-context.js (hunk 2 above)
Comment (Developers-codz) [reply src/context/cart-context.js]: yep

**Pull request merged**